mlflow/pyfunc/model.py

```diff
@@ -114,8 +118,25 @@ def __init__(self, func, hints=None, signature=None):
     def _get_type_hints(self):
         return _extract_type_hints(self.func, input_arg_index=0)
 
-    def predict(self, context, model_input):
-        return self.func(model_input)
+    def predict(self, context, model_input, params: Optional[Dict[str, Any]] = None):
+        """
+        :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
+                        can use to perform inference.
+        :param model_input: A pyfunc-compatible input for the model to evaluate.
+        :param params: Additional parameters to pass to the model for inference.
+
+                       .. Note:: Experimental: This parameter may change or be removed in a future
+                                               release without warning.
+
+        :return: Model predictions.
+        """
+        if params is None:
+            return self.func(model_input)
+        elif not isinstance(params, dict):
+            raise MlflowException(
+                f"Params must be a dictionary. Got type '{type(params).__name__}'."
+            )
+        return self.func(model_input, **params)
```

Comment:
Is this correct? I'd expect `func` to still take a dictionary of params, not keyword args.

cc @harupy

Reply:
`self.func` only accepts a single argument now.